### DISCUSSION TITLE
Spotube is playing songs not on my playlist

### DISCUSSION THREAD

**Discussion by clmaes** [KRTirtho/spotube#1339] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When i start to play numbers in my playlist and the song is finished the next song is not on my playlist. 

### Expected Behavior

When i start my playlist only the songs in that playlist should be playing

### Steps to reproduce

1. open spotube
2. choose an playlist
3. start the playlist

### Operating System

linux mint lmde 5

### Spotube version

3.5.0

### Installation source

Flathub (Flatpak)

### Additional information

Installed from the Program manager in mint

### Self grab

- [ ] I'm ready to work on this issue!

**Comment by ghost** [comment]
It's the "Endless Playback" function.

Turn it off in the settings (in the "Playback" section, the last position)

Then clear your current queue by clicking your current playback view (you can get it by playing a playlist if you don't have it above your menu buttons), then pressing on the "Queue" button and finally clicking at "Clear all".

Now play your playlist again and you should be free from that problem :D

**Comment by ajsyhurhdiu** [comment]
![spotube](https://github.com/user-attachments/assets/5ce073b3-9a45-42a1-988a-478849a6dfbf)
Change your setting to these, working perfectly after this, If this is also not working use proxies, It seems that there is a change in YouTube APIs and currently spotube is outdated because of it. I hope this solves your issue.

**Reply by ajsyhurhdiu** [reply]
Also Logout of your account and again enter the cookies sp_dc values again, If the issue doesnt resolves, lets hope that spotube developers delve into this issue and solve it, period

**Comment by ajsyhurhdiu** [comment]
So after I investigated the issue, it seems that YouTube and other streaming servers are working fine, issue comes from spotube itself, the Ip address of where this all data is stored, Spotube and other browser can't find the route, this usually happens when default gateway isn't configured correctly or the next-hop Ip address of our static route points to the correct router interface isn't placed correctly, so probably the issue lies in the default gateway configuration.
